core/templates/dev/head/pages/exploration-player-page/services/exploration-player-state.service.ts

```diff
@@ -82,11 +82,12 @@ angular.module('oppia').factory('ExplorationPlayerStateService', [
         returnDict, arePretestsAvailable, callback) {
       StateClassifierMappingService.init(returnDict.state_classifier_mapping);
       StatsReportingService.initSession(
-        explorationId, returnDict.exploration.title, version,
+        explorationId, returnDict.exploration.title, returnDict.version,
```

Comment:
One question -- is it possible to add a unit test for this line that fails with the old code and that passes with the new code?

Reply:
Hm, do you mean something like this:
In `exploration-player-state.service.spec.ts` have initializeExplorationServices pass returnDict.version = null and assert failure (and the success case)?

Reply:
Nope, although that's not a bad test either :)

I meant more like: write a single test for which the current code passes. But then, if you changed this back to the old code, the test you wrote should fail.

Reply:
Hi @seanlip, I tried to write a spec file for exploration-player-state.service but got this error:

Error: Trying to get the Angular injector before bootstrapping the corresponding Angular module.

I think this is because this file is using Angular and AngularJS dependencies. I tried injecting all the required deps but it still seems to say ExplorationPlayerStateService is null. I've pushed a dummy test so I can get some feedback and pointers on how to fix this.

Reply:
Can you get some guidance from @bansalnitish, @ankita240796 or @Showtim3? (This seems related to the work the Angular migration team is doing.)

Reply:
https://github.com/oppia/oppia/pull/8235#discussion_r360686858

Reply:
Thank you @ankita240796 for you comment.

@kevintab95 let us know if it still doesn't works.

Reply:
Thanks guys, using mocks instead of trying to inject all the deps fixed it! :)

Reply:
That's great! You're welcome @kevintab95!